ql/src/java/org/apache/hadoop/hive/ql/optimizer/GenMapRedUtils.java

```diff
@@ -1352,7 +1371,8 @@ public static void createMRWorkForMergingFiles(FileSinkOperator fsInput,
     Task<MoveWork> mvTask = GenMapRedUtils.findMoveTaskForFsopOutput(mvTasks, fsopPath, fsInputDesc.isMmTable(),
         fsInputDesc.isDirectInsert(), fsInputDesc.getMoveTaskId(), fsInputDesc.getAcidOperation());
     ConditionalTask cndTsk = GenMapRedUtils.createCondTask(conf, currTask, dummyMv, work,
-        fsInputDesc.getMergeInputDirName(), finalName, mvTask, dependencyTask, lineageState);
+        changeToTmpPath ? Utilities.toTempPath(fsInputDesc.getMergeInputDirName()) : fsInputDesc.getMergeInputDirName(),
```

Comment:
could you exract `changeToTmpPath ? Utilities.toTempPath(fsInputDesc.getMergeInputDirName()) : fsInputDesc.getMergeInputDirName()` this to a `getMergeInputDirName()` function?

Reply:
Done.